pkg/scheduler/apis/config/validation/validation.go

```diff
@@ -216,3 +217,115 @@ func validateExtendedResourceName(name v1.ResourceName) []error {
 	}
 	return validationErrors
 }
+
+// ValidateInterPodAffinityArgs validates that InterPodAffinityArgs are correct.
+func ValidateInterPodAffinityArgs(args config.InterPodAffinityArgs) error {
+	return ValidateHardPodAffinityWeight(field.NewPath("hardPodAffinityWeight"), args.HardPodAffinityWeight)
+}
+
+const (
+	// MinHardPodAffinityWeight is the minimum HardPodAffinityWeight.
+	MinHardPodAffinityWeight int32 = 0
+	// MaxHardPodAffinityWeight is the maximum HardPodAffinityWeight.
+	MaxHardPodAffinityWeight int32 = 100
+)
+
+// ValidateHardPodAffinityWeight validates that weight is within allowed range.
+func ValidateHardPodAffinityWeight(path *field.Path, w int32) error {
+	if w < MinHardPodAffinityWeight || w > MaxHardPodAffinityWeight {
+		msg := fmt.Sprintf("not in valid range [%d-%d]", MinHardPodAffinityWeight, MaxHardPodAffinityWeight)
+		return field.Invalid(path, w, msg)
+	}
+	return nil
```

Comment:
something similar can be done from those plugins, 
https://github.com/kubernetes/kubernetes/blob/7ba332a839c844ef4ba0191ddca2ce3601d283af/pkg/scheduler/framework/plugins/noderesources/most_allocated.go#L74-L79

https://github.com/kubernetes/kubernetes/blob/7ba332a839c844ef4ba0191ddca2ce3601d283af/pkg/scheduler/framework/plugins/noderesources/least_allocated.go#L75-L80

Reply:
except that not exported helper.

You could also do that in a follow up PR, but don't forget to list those plugins as not yet refactored in this PR description.

Reply:
I will handle Node Resources plugins in a separate PR as they are more complicated.